File: codes/Garnetwzy/102.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ret;
        dfs(ret, 0, root);
        return ret;
    }
    
    void dfs(vector<vector<int>>& ret, int height, TreeNode *root) {
        if(root == NULL)
            return;
        if(height == ret.size()) {
            ret.push_back(vector<int>());
        }
        ret[height].push_back(root->val);
        dfs(ret, height+1, root->left);
        dfs(ret, height+1, root->right);
    }
};

File: codes/HellDream/102.java
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> lists = new ArrayList<>();
        if(root==null) return lists;
        Queue<TreeNode> q = new ArrayDeque<>();
        q.offer(root);
        while(!q.isEmpty()){
            List<Integer> list = new ArrayList<>();
            int size = q.size();
            for(int i=0;i<size;i++){
                TreeNode node = q.poll();
                list.add(node.val);
                if(node.left!=null) q.offer(node.left);
                if(node.right!=null) q.offer(node.right);
            }
            lists.add(list);
        }
        return lists;
    }
}